src/tck/src/SdkClient.cc

```diff
@@ -39,21 +57,170 @@ constexpr auto DEFAULT_TCK_REQUEST_TIMEOUT = std::chrono::seconds(30);
 // The SDK Client to use to submit requests to a Hedera network.
 Client mClient; // NOLINT
 
+// Generate a Key object from an undetermined key DER-encoded hex string.
+std::shared_ptr<Key> getKey(const std::string& key)
+{
+  try
+  {
+    return PublicKey::fromStringDer(key);
+  }
+  catch (const BadKeyException&)
+  {
+    try
+    {
+      return PrivateKey::fromStringDer(key);
+    }
+    catch (const BadKeyException&)
+    {
+      proto::Key protoKey;
+      protoKey.ParseFromString(internal::Utilities::byteVectorToString(internal::HexConverter::hexToBytes(key)));
+      return Key::fromProtobuf(protoKey);
+    }
+  }
+}
+
+// Helper function used to generate a Key.
+std::string generateKeyHelper(const TckKey& key)
+{
+  // No type indicates a random ED25519/ECDSAsecp256k1 private/public key.
+  if (!key.mType.has_value())
+  {
+    // If no type is specified, nothing else should be specified.
+    if (key.mFromKey.has_value() || key.mThreshold.has_value() || key.mKeys.has_value())
+    {
+      throw JsonRpcException(JsonErrorType::INVALID_REQUEST,
+                             "invalid request: no other parameters should be specified if type isn't specified.");
+    }
+
+    // Generate a random key.
+    switch (internal::Utilities::getRandomNumber(0, 3))
```

Comment:
Maybe we can have some enumeration for the key type cases like:
```
enum class KeyType {
    ED25519Private,
    ED25519Public,
    ECDSAsecp256k1Private,
    ECDSAsecp256k1Public,
    KeyTypeCount
};
```

Reply:
Yep that makes sense, added